packages/solid-query/src/createBaseQuery.ts

```diff
@@ -159,8 +186,10 @@ export function createBaseQuery<
             unsubscribe = createClientSubscriber()
           }
         }
+
         if (!state.isLoading) {
-          resolve(state)
+          const query = observer.getCurrentQuery()
+          resolve(hydrateableObserverResult(query, state))
```

Comment:
Here is the code path that wasn't covered - this is triggered if the data is already pre-loaded prior to `createQuery`. We need to "sanitize" the state on the server before resolving, so that it can be correctly hydrated back to the client.